core/src/main/java/bio/terra/pearl/core/service/survey/SurveyParseUtils.java

```diff
@@ -144,4 +149,59 @@ public static String getUpstreamStableId(JsonNode calculatedValue) {
         return matcher.matches() ? matcher.group(1).trim() : null;
     }
 
+    /**
+     * This method is used to get the answer to a question by its stableId
+     * The method will return the answer as the specified class in returnClass
+     * The method will return null if the questionStableId is not found in the surveyJsonData
+     * @param surveyJsonData the survey json data
+     *                       The method assumes the surveyJsonData is a valid survey json of an array of questions with different stableIds
+     * @param questionStableId the stableId of the question to get the answer for
+     * @param returnClass the class to return the answer as
+     *                    The method assumes the returnClass is a valid class that can be used to convert the answer to
+     * @param objectMapper the object mapper to use to convert the answer to the returnClass
+     * @param <T> the class to return the answer as
+     * @return the answer to the question with the stableId questionStableId as the class returnClass
+     * */
+    public static <T> T getAnswerByStableId(String surveyJsonData, String questionStableId, Class<T> returnClass,
+                                            ObjectMapper objectMapper, String answerField) {
+        try {
+            JsonNode rootNode = objectMapper.readTree(surveyJsonData);
+            if (rootNode.isArray()) {
+                for (JsonNode node : rootNode) {
+                    if (questionStableId.equals(getQuestionStableId(node))) {
+                        return convertQuestionAnswerToClass(node, answerField, returnClass, objectMapper);
+                    }
+                }
+            }
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        return null;
+    }
+
+    /**
+     * This method is used to convert the answer to a survey to the returnClass. The method assumes the answer is a valid and is
+     * in the answerField of the node
+     * */
+    static <T> T convertQuestionAnswerToClass(JsonNode node, String answerField, Class<T> returnClass, ObjectMapper objectMapper) throws JsonProcessingException {
```

Comment:
nit: we generally include the visibility modifier -- I find it makes the code easier to read.
```suggestion
    protected static <T> T convertQuestionAnswerToClass(JsonNode node, String answerField, Class<T> returnClass, ObjectMapper objectMapper) throws JsonProcessingException {
```